dataStructures/queues/queue_test.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestNewOrder(t *testing.T) {
+	cases := []struct {
+		name, product, customerName string
+		priority, quantity          int
+		want                        *Order
+	}{
+		{"create new order", "macbook", "andela", 1, 20, &Order{1, 20, "macbook", "andela"}},
+		{"create another order", "windows", "apple", 2, 30, &Order{2, 30, "windows", "apple"}},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// In tests it's common to call the value that you're testing "got"
+			// If there are lots of them then "gotOrder" "gotQueue" etc. but in this test it doesn't matter
+			// We do it because then someone reading the code immediately knows that this is the value being tested
+			got := newOrder(tc.priority, tc.quantity, tc.product, tc.customerName)
+			if got.customerName != tc.want.customerName {
+				t.Errorf("newOrder(%d, %d, %q, %q) got +%s; want +%s", tc.priority, tc.quantity, tc.product, tc.customerName, got.customerName, tc.want.customerName)
+			}
+
+			if got.product != tc.want.product {
+				t.Errorf("newOrder(%d, %d, %q, %q) got +%s; want +%s", tc.priority, tc.quantity, tc.product, tc.customerName, got.product, tc.want.product)
+			}
+
+			if got.priority != tc.want.priority {
+				t.Errorf("newOrder(%d, %d, %q, %q) got +%v; want +%v", tc.priority, tc.quantity, tc.product, tc.customerName, got.priority, tc.want.priority)
+			}
+
+			if got.quantity != tc.want.quantity {
+				t.Errorf("newOrder(%d, %d, %q, %q) got +%v; want +%v", tc.priority, tc.quantity, tc.product, tc.customerName, got.quantity, tc.want.quantity)
+			}
+
+			if !reflect.DeepEqual(got, tc.want) {
```

Comment:
At Google we have two versions of [this document](https://github.com/golang/go/wiki/CodeReviewComments), the extra one being for comments about tests. One of them is "Compare full structures". You don't gain anything from individual checking every single field (and it makes your code hard to read _and_ write). 

If you use `[cmp.Diff()](https://godoc.org/github.com/google/go-cmp/cmp#Diff)` instead of `reflect.DeepEqual()` then you get the same effect.

```Go
if diff := cmp.DIff(tc.want, got); diff != "" {
  t.Errorf("… -want +got: %s", diff)
}
```

You'll notice that I swapped `want` and `got` around, and diffs are the only place that this is idiomatic in Go. The reason is that it then prints out a nicer message with `-` in front of everything that is missing from what you wanted, and `+` in front of everything that is added in what you got. A lot of idiomatic Go is about simple changes like this that make your code (and error messages) very easy for everyone else to use.